### PR TITLE
refactor(test): `thenify` not needed for `getVerificationLink`

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Replaced a lot of direct calls to `getVerificationLink` => `openPage`
with calls to `openVerificationLinkInSameTab`.

Modernized several modules while digging in:
* complete_sign_up
* oauth_sign_up

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r?

File: tests/functional/complete_sign_up.js [hunk 1]
```diff
@@ -5,50 +5,45 @@
...
+  var click = FunctionalHelpers.click;
   var createRandomHexString = TestHelpers.createRandomHexString;
+  var createUser = FunctionalHelpers.createUser;
   var fillOutSignUp = FunctionalHelpers.fillOutSignUp;
+  var getVerificationLink = FunctionalHelpers.getVerificationLink;
   var noSuchElement = FunctionalHelpers.noSuchElement;
+  var openPage = FunctionalHelpers.openPage;
+  var testElementExists = FunctionalHelpers.testElementExists;
   var testSuccessWasShown = FunctionalHelpers.testSuccessWasShown;
+  var visibleByQSA = FunctionalHelpers.visibleByQSA;
```
Comment (philbooth) [inline tests/functional/complete_sign_up.js]: When so many of these are changing at once, maybe it's a good moment to switch to destructuring assignment?

File: tests/functional/reset_password.js [hunk 2]
```diff
@@ -100,8 +102,8 @@ define([
       .then(openPage(url, header));
   });
 
-  function testAtSettingsWithVerifiedMessage(context) {
-    return context.remote
+  var testAtSettingsWithVerifiedMessage = thenify(function() {
```
Comment (philbooth) [inline tests/functional/reset_password.js]: What are our plans for `var`, generally? If we have a plan to one day lint them out of existence maybe `const` this instead?

**Comment by philbooth** [comment]
Looks good, r+.

File: tests/functional/complete_sign_up.js (hunk 1 above)
Comment (shane-tomlinson) [inline tests/functional/complete_sign_up.js]: Destructuring doesn't work by default with node 4.6. :/ We can `const` things though. I'll open up an issue about it.

**Pull request merged**